src/bgw/job.h

```diff
@@ -27,6 +27,14 @@ typedef struct BgwJob
 	BgwJobHistory job_history;
 } BgwJob;
 
+/* Positive result numbers reserved for success */
+typedef enum JobResult
+{
+	JOB_FAILURE_TO_START = -1,
+	JOB_FAILURE = 0,
```

Comment:
Maybe we should rename it to `JOB_FAILURE_IN_EXECUTION` or something, to make it absolutely clear that this is not a generic error code, and failure to start has its own code.

Reply:
Done!